src/components/follow-link/follow-link.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { Link, SystemIcon, ESystemIconNames } from '@mdb/flora';
+import { useSession } from 'next-auth/react';
+import useSignInURL from '../../hooks/sign-in-url';
+import { Tag } from '../../interfaces/tag';
+import Tooltip from '../tooltip/tooltip';
+
+interface FollowLinkProps {
+    topic: Tag;
+    iconsOnly?: boolean;
+}
+
+const FollowLink: React.FunctionComponent<FollowLinkProps> = ({
+    topic,
+    iconsOnly = false,
+}) => {
+    const { status, data: session } = useSession();
+
+    const [showClickTooltip, setShowClickTooltip] = useState(false);
+
+    const [timeoutID, setTimeoutID] = useState<NodeJS.Timeout | null>(null);
+
+    const signInURL = useSignInURL();
+
+    const isLoggedIn = status === 'authenticated';
+    const followedTopics = session?.followedTopics as Tag[] | undefined;
+    const isFollowingAnyTopics = !!followedTopics && !!followedTopics.length;
+    const isFollowing = useMemo(
+        () =>
+            !!(
+                followedTopics &&
+                followedTopics.find(
+                    followedTopic => followedTopic.slug === topic.slug
+                )
+            ),
+        [followedTopics, topic.slug]
+    );
+
+    const hoverTooltipText = useMemo(() => {
+        if (!isLoggedIn) return 'Sign in to follow topics';
+
+        if (!isFollowingAnyTopics)
+            return 'Receive a monthly digest and recommended content based on topics you follow!';
+
+        if (iconsOnly)
+            return isFollowing ? 'Unfollow this topic' : 'Follow this topic';
+    }, [isLoggedIn, isFollowingAnyTopics, iconsOnly, isFollowing]);
+
+    // If we're not in a hoverTooltipText scenario, there should be no hover tooltip rendered.
+    const hasHoverTooltip = !!hoverTooltipText;
+
+    const onFollowClick = useCallback(() => {
+        let topics: Tag[];
+        // Never show the click tooltip when using the iconsOnly variant
+        if (!isFollowingAnyTopics) {
+            // TODO: Enter modal flow
+            alert('Onboarding modal will go here');
+            return;
+        }
+        if (!iconsOnly) {
+            if (timeoutID) {
+                // To prevent a follow and unfollow action in under 2 seconds from having it's click tooltip dissapear prematurely.
+                clearTimeout(timeoutID);
+            }
+            setShowClickTooltip(true);
+            const id = setTimeout(() => setShowClickTooltip(false), 2000);
+            setTimeoutID(id);
+        }
+
+        if (followedTopics && isFollowing) {
+            topics = followedTopics.filter(
+                followedTopic => followedTopic.name !== topic.name
+            );
+        } else {
+            topics = followedTopics ? [...followedTopics, topic] : [topic];
+        }
+        console.log(topics); // TODO: POST followedTopics to external API endpoint.
```

Comment:
the modal will handle the selected topics and posting to the backend, assuming all of the PUT bodies will look the same